src/animation/interpolants/LinearInterpolant.js

```diff
@@ -0,0 +1,44 @@
+/**
+ *
+ * Interpolant the returns a time-proportional mix of the keyframe values of
+ * the surrounding interval.
+ *
+ *
+ * @author tschw
+ */
+
+THREE.LinearInterpolant = function( times, values, stride, result ) {
+
+	THREE.Interpolant.call( this, times, values, stride, result );
+
+};
+
+Object.assign( THREE.LinearInterpolant.prototype, THREE.Interpolant.prototype, {
+
+	constructor: THREE.LinearInterpolant,
+
+	_interpolate: function( i1, t0, t, t1 ) {
```

Comment:
Well, in this (pretty much mathematical) context, the names are descriptive. You just proved it by guessing them right :-). The reason I used them is they are concise and common, and I don't really like very long lines in source files, as resulting from using long speaking names in mathematical formulae: They make files harder to read, edit, merge and debug.

I used `keyTime` and `prevKeyTime` in higher level code, making it clear what (just, "unsubscripted") `index` refers to...

The `Interpolant` family of classes does not depend on animation stuff. In fact, we're one layer below. Therefore we're not *really* talking about "times" and "keyframes" or anything "current", but about a parameter domain, commonly using the variable `t` in literature, subscripted for constants in respect to a line segment or an entire curve (depending on the context) and indices of samples, commonly `i` in literature, IOW "times" and "keyframes" are *only* the context-dependent actuals at the call site, *not really* the formal ones;l `t` may also be the parameter of a parametric curve that defines its tessellation and the samples are in fact "control points" in this context, other possible context include numeric methods (fully abstract), audio ("samples" on a time domain, its scale being the pitch)... 

I think we should move the interpolants out of the animation folder. Maybe to math. Also seems to go well your suggestion to use them for curves. 

Opinions anyone?